packages/react-list/Spec.md

```diff
@@ -2,62 +2,375 @@
+implication that v8, but is still fully customizable if needed.
+
+Grid layout is a scenario v8 has documented that is not supported by the v0 `List` at all.
+
+Compact layout is also possible with `DetailsList`, which reduces the extra spacing between list items and removes
+padding with the boundary of the list.
+
+v8 selection scenarios for `List` components will use a checkbox for each item, whereas v0 will use background color
+changes to indicate selection.
+
+#### Virtualization
+
+V8 list components all support virtualization internally by default. This means that no extra work needs to be
+done by consumers to have virtualized lists, but the virtualization code will be present even if the use case
+does not need it. However, the virtualization functionality is only implemented once in the `List` component. All
+v8 list variants will reuse the `List` component to virtualize their elements.
+
+v0 Does not offer virtualization built into any component. The library manages a collection of prototypes on the
+docsite that show users how to virtualize v0 components:
+
+- [Virtualized tree](https://fluentsite.z22.web.core.windows.net/0.61.0/virtualized-tree)
+- [Virtualized sticky tree](https://fluentsite.z22.web.core.windows.net/0.61.0/virtualized-sticky-tree)
+- [Virtualized table](https://fluentsite.z22.web.core.windows.net/0.61.0/virtualized-table)
+
+The `List` component in v0 does not have such a prototype all popular virtualization libraries use lists as the simplest example.
+
+#### Keyboard navigation
+
+Both libraries allow keyboard navigation for their list components.
+
+v8 list components enable keyboard navigation without any changes to the underlying DOM elements.
+
+v0 `List` component enables keyboard navigation but changes the semantics of the component to be a menu with menuitems
+instead.
+
+#### Imperative methods
+
+v8 exposes an `IList` interface of imperative methods through a component ref.
+
+```tsx
+const Sample = () => {
+  const ref = React.useRef({});
+  // ref.forceUpdate()
+  // ref.getStartItemIndexInView()
+  // ref.scrollToIndex()
+  // ref.getTotalListHeight()
+
+  return <BasicList componentRef={ref}>;
+}
+```
 
 ## Sample Code
 
-_Provide some representative example code that uses the proposed API for the component_
+Presentational/Non-interactive list
+
+```tsx
+<List>
+  <ListItem>List Item</ListItem>
+  <ListItem>List Item</ListItem>
+  <ListItem>List Item</ListItem>
+  <ListItem>List Item</ListItem>
+</List>
+```
+
+Selectable listbox
+
+```tsx
+<Listbox>
+  <ListboxOption value="1">Option 1</ListboxOption>
+  <ListboxOption value="2">Option 2</ListboxOption>
+  <ListboxOption value="3">Option 3</ListboxOption>
+  <ListboxOption value="4">Option 4</ListboxOption>
+</Listbox>
+```
+
+Selectable listbox without checkbox indicator slot
+
+```tsx
+<Listbox>
+  <ListboxOption checkbox={null} value="2">
```

Comment:
this API feels a bit strange IMO.  I see it's type annotated as `Slot<'span'>` thus the nullability. Do we want to actually allow user to specify any kind of magic html (which might not end up being a checkbox) to pass to this property ?